src/detect-engine-threshold.c

```diff
@@ -713,6 +658,31 @@ void ThresholdHashInit(DetectEngineCtx *de_ctx)
     }
 }
 
+/**
+ * \brief Realloc threshold context hash tables
+ *
+ * \param de_ctx Detection Context
+ */
+void ThresholdHashRealloc(DetectEngineCtx *de_ctx)
+{
+    /* Return if we are already big enough */
+    uint32_t num = de_ctx->signum + 1;
+    if (num <= de_ctx->ths_ctx.th_size)
+        return;
+
+    void *ptmp = SCRealloc(de_ctx->ths_ctx.th_entry, num * sizeof(DetectThresholdEntry *));
+    if (ptmp == NULL) {
+        SCLogWarning(SC_ERR_MEM_ALLOC, "Error allocating memory for rule thresholds"
```

Comment:
Please use a value prefixed with `SC_WARN_` .... These are in `util-error.[hc]`

Reply:
I actually went looking for an appropriate SC_WARN to use here, but none of them seem to apply to something like a memory allocation failure. The only ones I can find that apply to memory allocation problems are SC_ERR, and there are examples of SCLogWarning(SC_ERR_MEM_*) in conf.c, log-pcap.c, log-tlsstore.c, and util-buffer.c, so I thought this was fine.

Would you like a new SC_WARN_MEM_ALLOC in util-error.h?

Reply:
No need for that. We're considering switching to a much simpler scheme soon too.

Reply:
No ... your pr looks good to me as is.

We'll address the discrepancy in a different issue.